jetty-core/jetty-quic/jetty-quic-api/src/main/java/org/eclipse/jetty/quic/api/Session.java

```diff
@@ -0,0 +1,202 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.quic.api;
+
+import java.net.SocketAddress;
+import java.util.Collection;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.quic.api.frames.ConnectionCloseFrame;
+import org.eclipse.jetty.quic.api.frames.DataBlockedFrame;
+import org.eclipse.jetty.quic.api.frames.MaxDataFrame;
+import org.eclipse.jetty.quic.api.frames.MaxStreamsFrame;
+import org.eclipse.jetty.quic.api.frames.StreamsBlockedFrame;
+import org.eclipse.jetty.quic.api.frames.TransportParameters;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.Promise;
+
+/**
+ * <p>Represents a QUIC connection to a remote peer.</p>
+ * <p>A {@link Session} represents the active part of the connection, and by calling its APIs
+ * applications can generate events on the connection.</p>
+ * <p>Conversely, {@link Session.Listener} is the passive part of the connection,
+ * and has callback methods that are invoked when events happen on the connection.</p>
+ */
+public interface Session
+{
+    /**
+     * @return the QUIC connection id
+     */
+    String getId();
```

Comment:
why not a `TransportParameter.Id`?  If not appropriate, why is that class public in the API?

Reply:
`Session.id` is the unique connection id for that session, and is not related to `TransportParameters`.

`TransportParameters.Id` is public because applications should be able to configure transport parameters and possibly create custom ones.